### DISCUSSION TITLE
Importing CSVs: is it possible to change source type of a column?

### DISCUSSION THREAD

**Discussion by iuliana-laurentia** [dbeaver/dbeaver#15484] [Q&A]
Hi Team,

I am trying to import a csv on  postgresql using DBeaver. However, I am getting issues as the source datatype for **elementId** columns is being set automatically  to **int**, and this causes problems with uploading. The format of the elementId column in the csv file is set to **General**.

Below are some screenshot of the steps I am taking: 
![image](https://user-images.githubusercontent.com/51965110/153851274-75516ab7-30f4-4924-8663-7496bd00fe37.png)

![image](https://user-images.githubusercontent.com/51965110/153851395-df9237cf-7d16-4cb0-a58e-1acc19b41bea.png)
 
![image](https://user-images.githubusercontent.com/51965110/153851488-2da9ba01-79b7-4f76-849e-1510d01870da.png)

The **elementId** column should be **varchar** instead of **int**. Is there any way of changing the source type?  

Thanks a lot! Any advice will be highly appreciated!

**Comment by LonwoLonwo** [comment]
Hello @iuliana-laurentia 

Have you tried to change int type to varchar in the target data type column?

![2022-02-14 17_08_53-Importing CSVs_ is it possible to change source type of a column_ · Discussion #](https://user-images.githubusercontent.com/45152336/153880509-5367b1ed-27e2-470d-a287-a3ab6f3325e2.png)

You can check the result of this operation in the "Target DDL" button.

![2022-02-14 17_10_53-DBeaver Ultimate 21 3 5 - employee_insert_trigger](https://user-images.githubusercontent.com/45152336/153880526-aac4a799-74e9-4152-8162-233bab22d6c4.png)

**Comment by liar666** [comment]
Solution does not work 
- for an Oracle Backend
- With DBeaver Version 23.3.0.202312031432

Simple reproduction of the error:
- Download CSV from (as of 2023/12/13):
https://opendata.paris.fr/explore/dataset/les-arbres/export/
- Try to import (needs to change delimiter to ";").
- Get error "SQL Error [12899] [72000]: ORA-12899: value too large for column "C##GUILLAUME"."LES_ARBRES"."COMPLEMENT ADRESSE""
- "COMPLEMENT ADRESSE"" is detected as 'Source Type = VARCHAR2(50)' [which cannot be changed] and changing "Target Type" to anything does not help...

Same problem occurs with column "VARIETE OUCULTIVAR". Ignoring the 2 columns makes the import work, but that's not what one needs...

**Reply by LonwoLonwo** [reply]
You can change your target column data types to any other correct format on the Tables Mapping page.

**Reply by liar666** [reply]
That's exactly what I describe I do, but it's the "Source Type" that causes problem... changing the "Target Type" does not help

**Reply by LonwoLonwo** [reply]
How exactly does the "Source Type" cause the issue? Because it is just an info label, it doesn't have an impact. You can check your target type in the DDL section (if columns are new in a destination table). It would help if you increased the length of your target type. If you are exporting in the new table. If your table already exists - then change your table data type first.

**Reply by liar666** [reply]
Seems images will be simpler to explain the problem:
Source type is detected to VARCHAR2(50) for "COMPLEMENT ADRESSE". I Force the Target Type to be unbounded VARCHAR2
![screenshot_2023-12-13-18 47](https://github.com/dbeaver/dbeaver/assets/3216927/0cd95485-b973-4ee0-ba76-0a2bceb4f6e5)
The error is that is it unsufficient to store the data in this column:
![screenshot_2023-12-13-18 48](https://github.com/dbeaver/dbeaver/assets/3216927/3ef4b78e-18c3-459f-9980-efb191ac026c)

**Reply by LonwoLonwo** [reply]
But you can't create a VARCHAR2 column without a length (usually). Why can't you just increase the data type length? Till 128, as example. Or 54.

It worked for me:

![2023-12-13 20_04_06-DBeaver Ultimate 23 3 0 - LES_ARBRES](https://github.com/dbeaver/dbeaver/assets/45152336/844c3754-45b0-49af-9b4c-6eafc258a084)

**Reply by liar666** [reply]
Doesn't work for me with DBeaver 23.3.0.202312031432, that why I created the issue from the beginning.

Whatever value I set in Target Type, I get the error.

**Reply by liar666** [reply]
![screenshot_2023-12-13-19 09](https://github.com/dbeaver/dbeaver/assets/3216927/d3947eac-9c92-418b-a576-6957511a8abf)
![screenshot_2023-12-13-19 11](https://github.com/dbeaver/dbeaver/assets/3216927/7b41cc23-d494-431e-81b5-ecdb91601e0c)

**Reply by LonwoLonwo** [reply]
Could you please provide a screen from the Target DDL tab?

**Reply by liar666** [reply]
Here it is...

![screenshot_2023-12-14-08 18](https://github.com/dbeaver/dbeaver/assets/3216927/97815b73-0535-4859-ab93-4eb81e228c00)

**Reply by LonwoLonwo** [reply]
No changes mean - the table already exists.

>  If your table already exists - then change your table data type first.

So, please, try to change the existing data type of your table before data transferring. Not during data transferring. 
Or try to transfer your data to the new table - with the "create" status on the table mapping page.

**Comment by i327776** [comment]
Dear beaver, please do not reinvent the wheel. My file has 1 million rows, all of them of numeric type, except the last one which is an NVARCHAR. Let me manually change the datatype of the source, after all, it's my file, not yours!!!!
